modules/boss/src/main/ts/ephox/boss/mutant/Query.ts

```diff
@@ -9,6 +9,8 @@ const extract = function (item: Gene): string[] {
   return [ self ].concat(rest);
 };
 
+// TODO: This is broken. See TINY-6501, but the gist is that the behaviour of this function should match
```

Comment:
Is this really broken? It works for the use case, it's just not a complete implementation. Maybe something like this instead?
```suggestion
// TODO: This is only partially implemented. See TINY-6501, but the gist is that the behaviour of this function should match
```